modules/tile-converter/src/3d-tiles-converter/helpers/load-i3s.ts

```diff
@@ -40,13 +49,83 @@ export const loadI3SContent = async (
       },
       _tilesetOptions: {
         store: sourceTileset.store,
+        // @ts-expect-error
         attributeStorageInfo: sourceTileset.attributeStorageInfo,
+        // @ts-expect-error
         fields: sourceTileset.fields
       }
     }
   };
-  const tileContent = await load(sourceTile.contentUrl, I3SLoader, loadOptions);
+  const tileContent = await loadFromArchive(
+    sourceTile.contentUrl,
+    I3SLoader,
+    loadOptions,
+    slpkFilesystem
+  );
 
-  // @ts-expect-error
   return tileContent;
 };
+
+/**
+ * Load local SLPK file to ZipFileSystem instance
+ * @param url - path to SLPK file
+ * @returns instance of ZipFileSystem or null if url is not an SLPK file
+ */
+export async function openSlpk(url: string): Promise<ZipFileSystem | null> {
+  const slpkUrlParts = url.split('.slpk');
+  const {slpkFileName} = getSlpkUrlParts(url) || {};
+  if (slpkFileName) {
+    const slpkFileName = `${slpkUrlParts[0]}.slpk`;
+    const fileProvider = new FileHandleFile(slpkFileName);
+    const archive = await parseSLPKArchive(fileProvider, undefined, slpkFileName);
+    const fileSystem = new ZipFileSystem(archive);
+    return fileSystem;
+  }
+  return null;
+}
+
+/**
+ * Load a resource with load options and .3tz format support
+ * @param url - resource URL
+ * @param loader - loader to parse data (Tiles3DLoader / CesiumIonLoader)
+ * @param loadOptions - i3s loader options
+ * @returns i3s resource
+ */
+export async function loadFromArchive(
+  url: string,
+  loader: LoaderWithParser,
+  loadOptions: I3SLoaderOptions,
+  fileSystem: ZipFileSystem | null
+) {
+  const slpkUrlParts = getSlpkUrlParts(url);
```

Comment:
Maybe this path processing could be resolved inside an IndexFileSystem.fetch()

Reply:
I use this function to detect if there is an archive. If there isn't an archive, FileSystem won't be created.